### ISSUE TITLE
How do I make a table where the cells are sized to the margin?

### ISSUE THREAD

**Issue by Michael-Huang-9001** [dolanmiu/docx#92]
For example, I want to make a 4x4 table, and I want those 4x4 cells to be scaled to the size allowed by the margins, how do I do that?

Thanks.

**Comment by dolanmiu** [comment]
That is not supported yet

Will add as a feature request

**Comment by Michael-Huang-9001** [comment]
It seems that I can make them fit to the margins by setting the cell width to something large like 20000.

I have another question, how can I anchor a picture to a header/footer?
I can't seem to win against the margins at the top of the header/bottom of the footer.

Also, a funny thing, if I tried to insert a picture into the footer (haven't tried non-header/footer yet) and the path to the file or the file has a space in its name, it corrupts the output for some reason.

**Comment by Michael-Huang-9001** [comment]
It seems for my use case, I simply need the image in the header to be wrapped behind text and move to a specified location.

Is there anything that this module can do on that front?

**Comment by dolanmiu** [comment]
Do you mean fixed image positioning? And then placing that image behind the text?

If so, fixed positioning of images are coming soon! The code is there, just need to release

**Comment by Michael-Huang-9001** [comment]
Hi, thanks for being so helpful so far. This module is great.
I have one final small question.
My express packer is throwing out errors and when I call:

router.post('/', (req, res) => {
    ...
    const express_exporter = new docx.ExpressPacker(doc, res);
    express_exporter.pack('docx');
});

It works fine with a localpacker, but the express packer gives this error:

_http_outgoing.js:471
    throw new ERR_HTTP_HEADERS_SENT('set');
    ^

Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at ServerResponse.setHeader (_http_outgoing.js:471:11)
    at ServerResponse.header 
    at ServerResponse.send 
    at ServerResponse.ExpressPacker.res.on 
    at ServerResponse.emit (events.js:187:15)
    at emitCloseNT (_http_server.js:589:8)
    at process._tickCallback (internal/process/next_tick.js:63:19)

Any ideas? Thanks.

**Comment by Michael-Huang-9001** [comment]
Also, it seems the PDFpacker ignores any picture scaling I've made.

**Comment by dolanmiu** [comment]
It looks like you are calling some express related stuff along with pack()

Are you calling res.send(), or res.end() by any chance?

**Comment by Michael-Huang-9001** [comment]
I'm not calling those at all.

**Comment by Michael-Huang-9001** [comment]
Okay, I got the express packer spit out a download but I still cannot seem to get that error regarding the set header.

I just can't seem to get rid of that error and try/catch doesn't seem to stop it from crashing the Node process.

**Comment by Michael-Huang-9001** [comment]
Welp, I circumvented the express packer by using a local packer and downloading that file.

Thanks anyway, awesome module you've created! very neat. I was using Officegen but that one didn't have header/footer image support, many thanks for that.

:]

**Comment by Michael-Huang-9001** [comment]
Hmmm...
I think the header issue is because I called some next() and that causes it to call some middleware to call res.send or res.end somewhere.
Is it possible to not call res.end in the expresspacker?
Officegen had a streamer and I just set an attachment.

```
//Officegen's method
await docx.generate(res);
console.log("DOCX streamed to res.");
res.set({"Content-Type": "application/vnd.openxmlformats-officedocument.wordprocessingml.document", 'Content-disposition': 'attachment; filename=' + filename'});

```

**Comment by dolanmiu** [comment]
Good suggestion, will add in the major version of `docx` and this would be breaking

**Comment by simonmthiel** [comment]
+1 for this feature request

**Comment by dolanmiu** [comment]
Fixed in #288 

API will be like so:

https://github.com/dolanmiu/docx/blob/master/demo/demo32.ts#L59-L64

Will be released in v5
